synapse/config/ratelimiting.py

```diff
@@ -24,7 +24,7 @@ def __init__(
         defaults={"per_second": 0.17, "burst_count": 3.0},
     ):
         self.per_second = config.get("per_second", defaults["per_second"])
-        self.burst_count = config.get("burst_count", defaults["burst_count"])
+        self.burst_count = int(config.get("burst_count", defaults["burst_count"]))
```

Comment:
Because now mypy hates me. I spent a while trying to figure out how to make the type hints in `synapse/app/ratelimiting.py` accept floats or ints for burst count and I almost cried.

Anyway, it doesn't really make sense for burst count to be a float so :shrug:

Reply:
I'm surprised there isn't a `Number = Union[int, float]` 🤷 We could add that to `synapse.types` if it was useful.

Reply:
I recommend **not** looking at `numbers.Number`